draco/analysis/dayenu.py

```diff
@@ -655,20 +667,30 @@ def process(self, hv, source):
 
                     # Apply the filter
                     vis[pp, :, xx, :, tt] = np.matmul(NF, tvis)
-                    weight[pp, :, xx, tt] = tools.invert_no_zero(
-                        np.matmul(np.abs(NF) ** 2, tvar)
-                    )
-                    # Flag frequencies with large attenuation
-                    if self.atten_threshold > 0.0:
-                        diag = np.abs(np.diag(NF))
-                        nonzero_diag_flag = diag > 0.0
-                        if np.any(nonzero_diag_flag):
-                            med_diag = np.median(diag[nonzero_diag_flag])
-                            flag_low = diag > (self.atten_threshold * med_diag)
-                            weight[pp, :, xx, tt] *= flag_low.astype(weight.dtype)
+
+                    # Propagate the weights through the filter
+                    if not self.copy_weight:
+                        tvar = tools.invert_no_zero(weight[pp, :, xx, tt])
+
+                        weight[pp, :, xx, tt] = tools.invert_no_zero(
+                            np.matmul(np.abs(NF) ** 2, tvar)
+                        )
+
+                        # Flag frequencies with large attenuation
+                        if self.atten_threshold > 0.0:
+                            diag = np.abs(np.diag(NF))
+                            nonzero_diag_flag = diag > 0.0
+                            if np.any(nonzero_diag_flag):
+                                med_diag = np.median(diag[nonzero_diag_flag])
+                                flag_low = diag > (self.atten_threshold * med_diag)
+                                weight[pp, :, xx, tt] *= flag_low.astype(weight.dtype)
 
             self.log.debug(f"Took {time.time() - t0:0.3f} seconds in total.")
 
+        # If requested copy over the weight dataset
+        if self.copy_weight:
+            weight[:] = source.weight[:].local_array.copy()
```

Comment:
Is the `.copy` is needed here? Since you're writing into an existing and distinct array, I don't think that there would be any shared memory even without the copy.

Reply:
Good call.  I've removed this copy.